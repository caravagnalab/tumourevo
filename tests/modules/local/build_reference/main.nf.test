nextflow_process {
  name 'Test process BUILD_REFERENCE'
  script "modules/local/build_reference/main.nf"
  process 'BUILD_REFERENCE'

  test("Should run without failures") {
    when {
      process {
        """
        meta = [
            id : "idProva",
            tumour_sample: "Set6_42",
            cancer_type: "PANCANCER" 
        ]

        cds_table = file(params.testdata_base_path + '/DNDSCV/BUILD_REFERENCE/BioMart_human_GRCh37_chr3_segment.txt', checkIfExists: true)
        genome_fasta = file(params.testdata_base_path + '/DNDSCV/BUILD_REFERENCE/chr3_segment.fa', checkIfExists: true)
        input[0] = tuple(meta, cds_table, genome_fasta)
        """
      }
    }

    then {
      assert process.success
    }
  }
}