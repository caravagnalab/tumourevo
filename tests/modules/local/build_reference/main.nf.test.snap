{
    "Should run without failures": {
        "content": [
            [
                [
                    {
                        "id": "idProva",
                        "tumour_sample": "Set6_42",
                        "cancer_type": "PANCANCER"
                    },
                    "reference.rda:md5,7441c70e011b519107dc538689717e55"
                ]
            ]
        ],
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "23.10.1"
        },
        "timestamp": "2024-08-02T14:38:08.714375"
    }
}